### ISSUE TITLE
onInitialized event for iframe.

### ISSUE THREAD

**Issue by Pingguoren** [ariya/phantomjs#13994]
It looks like the onInitialized event is triggered only when the main page or the new window (through window.open or click on target=blank) is created. But it's not like that for the iframes.

Yeah, onResourceRequested works fine for iframes, therefore, allowing for headers substitution. But there is no way (at least I don't know one) to modify navigator properties through onResourceRequested/onResourceReceived/onPageLoad/onPageFinished/etc... The only way that I know and use is by means of onInitialized, which is not supported for iframes.

Here is the well-known example:

`
var onInitialized = function()
{
page.customHeaders = headers;
console.log("[page.onInitialized] " + this.url);
page.evaluate(function(options)
{
(function()
{
window.navigator.__defineGetter_('platform', function ()
{
return options.useragent["navigator.platform"];
});
....
})();
}, options);
};

page.onInitialized = onInitialized;
`

Is that possible somehow to make the frames use the navigator properties defined for the main window or to set them directly for each frame? Would not it be useful to have something like onFrameCreated (like onPageCreated) for each frame? For example, I use this code for new windows:

'
_onPageCreated = function(childPage)
{
console.log("[page.onPageCreated] " + childPage.url);

childPage.settings.userAgent = options.useragent["navigator.userAgent"];
childPage.customHeaders = headers;
childPage.viewportSize = { width:400, height: 500 };

childPage.onInitialized = _onInitialized;
childPage.onNavigationRequested = _onNavigationRequested;
childPage.onLoadStarted = _onLoadStarted;
childPage.onLoadFinished = _onLoadFinished;
childPage.onResourceRequested = _onResourceRequested;
childPage.onResourceReceived = _onResourceReceived;
childPage.onPageCreated = _onPageCreated;
};
'

It recursively defines event handlers for each new window. So, if the new window 2 is opened in the new window 1, it automatically inherites all handlers so i don't need to bother about my new windows.

I think the same should be made for the iframes/frames, because in each iframe/frame there could be another frame, and deep inside it another one and so on. If by the same recursive trick it would be possible to define all handlers, then all iframes/frames would be automatically accounted for. Especially, if we'll have onFrameInitialized for each frame, where it would be possible to define frame properties, such as window.navigator object.

To sum up, I'd like to know if there is anyway to control the navigator object of iframes/frames. If not, then I think that these feature should be added to CasperJS.
